### PR TITLE
refactor: rename get item batch to get item metadata batch and get item and fetch vectors batch to get item batch

### PR DESCRIPTION
**Pull request by malandis**
For clarity rename `get_item_batch` to `get_item_metadata_batch` and `get_item_and_fetch_vectors_batch` to `get_item_batch`.

### CODE REVIEW

File: src/momento/internal/aio/_vector_index_data_client.py [hunk 1]
```diff
@@ -196,53 +196,53 @@ async def get_item_batch(
...
             )
 
-            batch_response: vectorindex_pb._GetItemAndFetchVectorsBatchResponse = (
-                await self._build_stub().GetItemAndFetchVectorsBatch(request, timeout=self._default_deadline_seconds)
+            batch_response: vectorindex_pb._GetItemBatchResponse = await self._build_stub().GetItemBatch(
+                request, timeout=self._default_deadline_seconds
             )
-            self._log_received_response("GetItemAndFetchVectorsBatch", {"index_name": index_name})
-            return GetItemAndFetchVectorsBatch.Success.from_proto(batch_response)
+            self._log_received_response("GetItemBatch", {"index_name": index_name})
```
Comment (pgautier404) [inline src/momento/internal/aio/_vector_index_data_client.py:241]:
```suggestion
            self._log_received_response("GetItemMetadataBatch", {"index_name": index_name})
```

**Review by nand4011 — DISMISSED** [review]
LGTM, although it would be easy to miss things when Towers of Hanoiing function names like this.

File: src/momento/vector_index_client.py [hunk 2]
```diff
@@ -261,26 +261,24 @@ def get_item_batch(self, index_name: str, ids: list[str]) -> GetItemBatchRespons
         """Gets a batch of items from a vector index by ID.
 
         Args:
-            index_name (str): Name of the index to get the items from.
+            index_name (str): Name of the index to get the item from.
             ids (list[str]): The IDs of the items to be retrieved from the index.
 
         Returns:
-            GetItemBatchResponse: The result of a get item batch operation.
+            GetItemBatchResponse: The result of a get item and fetch vectors batch operation.
```
[the remark this answers is not in the document]
Comment (malandis) [reply src/momento/vector_index_client.py:268]: Not sure what you mean by the function names?

[the remark this answers is not in the document]
Comment (malandis) [reply src/momento/vector_index_client.py]: Nope that was a carryover by mistake. Thanks! Fixed in [a64273c](https://github.com/momentohq/client-sdk-python/pull/419/commits/a64273c82ee55be3c5b33db2a7297565b56e2888)

**Review by nand4011 — APPROVED** [review]

**Review by pgautier404 — APPROVED** [review]
LGTM

**Pull request merged**